src/soap/tests/test_client.py

```diff
@@ -59,6 +71,20 @@ def test_server_cert_specified(self):
             client.transport.session.verify, self.server_cert.public_certificate.path
         )
 
+    def _get_vcr_kwargs(self):
+        return {**super()._get_vcr_kwargs(), "record_mode": "none"}
+
+    def test_server_cert_pair(self):
+        # Should just use the certificate (chain) of the pair
+        service = SoapServiceFactory.build(
+            url=WSDL_URI, server_certificate=self.server_cert_and_privkey
+        )
```

Comment:
I agree.  The Client  code I was refactoring, had this behaviour. I just wanted to make sure my refactor did not change behaviour.
But also: I have failed to convince people to change their bad practises concerning private key material in the past. When the server is a gateway, it's imaginable that the one configuring the service has both halves of the key.

Reply:
> The Client code I was refactoring, had this behaviour. I just wanted to make sure my refactor did not change behaviour.

I think I remember that case and dropped it because it doesn't seem like a case we should have defined behaviour for.

> But also: I have failed to convince people to change their bad practises concerning private key material in the past. When the server is a gateway, it's imaginable that the one configuring the service has both halves of the key.

You are alluding that because they have both private and public key and they see that they can upload a private key, they will do so because they don't understand what is happening?

It's a risk, but in the market Open Forms is deployed in, it's sometimes bonkers how many different departments are involved in setting up such a connection.

Either way - I think this should be captured in the configuration documentation, can't hurt to point out there that you need to create a certificate record of type "cert only" and upload the public certificate.

As for this test - I'd either remove it to make the behaviour "undefined" or add a comment that this is not normal configuration, Open Forms should not have the private key of the server and that the observed behaviour in the test is not set in stone.